source/hp/dof_handler.cc

```diff
@@ -227,7 +227,7 @@ namespace internal
           vertex_fe_association (dof_handler.finite_elements->size(),
                                  std::vector<bool> (dof_handler.tria->n_vertices(), false));
 
-          for (typename DoFHandler<dim,spacedim>::active_cell_iterator
+          for (TriaActiveIterator < dealii::DoFCellAccessor < DoFHandler < dim, spacedim >, false > >
```

Comment:
This an all of the similar changes below I really resent. It makes the code impossible to read. We've worked hard to have a consistent set of typedefs with the local `*_iterator` types, and weakening this system in these ways in so many places makes it (i) impossible to ever change the actual data type that underlies these iterators, (ii) makes it impossible to read the code, (iii) makes it appear impossible to write code using one compiler that we can be confident will also work with MS VC.

I thought that you could compile the entire library already before. Can you elaborate on the reason why you need changes such as this one, including posting an error message or failure mode? Maybe we can come up with a better solution.